entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,128 @@
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToApply)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+CROSS APPLY [Posts] AS [p]
+
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], ([b].[Url] + N'=>') + [p].[Title] AS [p]
+FROM [Blogs] AS [b]
+OUTER APPLY [Posts] AS [p]
+```
+
+## GroupBy
+
+LINQ GroupBy operator creates result of type `IGrouping<TKey, TElement>` where `TKey` & `TElement` could be arbitrary types. Further, `IGrouping` implements `IEnumerable<TElement>`, which means you can compose over using any LINQ operator after grouping. Since no database structure can represent IGrouping, Group By operator has no translation in most cases. When an aggregate operator is applied to each group to which gives back a scalar, it can be translated to SQL `GROUP BY` in relational databases. SQL `GROUP BY` is restrictive too. It requires you to group by only scalar values  projection can contain only grouping key columns or aggregate applied over any column. EF Core identifies this subset and translates it to server. Following example shows a query, which has aggregate operator applied to it and generated SQL.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupBy)]
+
+```SQL
+SELECT [p].[AuthorId] AS [Key], COUNT(*) AS [Count]
+FROM [Posts] AS [p]
+GROUP BY [p].[AuthorId]
+```
+
+EF Core also translates when aggregate operator appears in Where filter by using `HAVING` clause in SQL and aggregate operator appears in Order By (or other ordering operators). The query before applying Group By operator can be any complex query using various different operators as long as it can be translated to server. Further, once you apply aggregate operator on grouping query to remove groupings from resulting source, you can compose it like any other query.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupByFilter)]
+
+```SQL
+SELECT [p].[AuthorId] AS [Key], COUNT(*) AS [Count]
+FROM [Posts] AS [p]
+GROUP BY [p].[AuthorId]
+HAVING COUNT(*) > 0
+ORDER BY [p].[AuthorId]
+```
+
+The aggregate operators EF Core supports are following
+
+- Average
+- Count
+- LongCount
+- Max
+- Min
+- Sum
+
+## Left Join
```

Comment:
This repeats the above GroupJoin section no?

Reply:
Nope.